### PR TITLE
tor: Change auth order to only use password auth if -torpassword

### PR DESCRIPTION
**Pull request by laanwj**
Change authentication order to make it more clear, and make it possible to use password authentication even though the Tor instance supports cookie auth (see #7700).

Before:
- If Tor is configured to allow cookie authentication, always try cookie authentication first. If cookie authentication fails, stop trying.
- Only if cookie authentication is not supported try password authentication with value of  `-torpassword` (if provided)

After:
- If the `-torpassword` option is provided, force use of password auth.
- Give error message if `-torpassword` provided, but password auth is not available.
- Give error message if only password auth available, but  `-torpassword` not given.

### CODE REVIEW

File: src/torcontrol.cpp [hunk 1]
```diff
@@ -595,13 +603,7 @@ void TorController::protocolinfo_cb(TorControlConnection& conn, const TorControl
...
             }
         } else if (methods.count("HASHEDPASSWORD")) {
-            if (!torpassword.empty()) {
-                LogPrint("tor", "tor: Using HASHEDPASSWORD authentication\n");
-                boost::replace_all(torpassword, "\"", "\\\"");
-                conn.Command("AUTHENTICATE \"" + torpassword + "\"", boost::bind(&TorController::auth_cb, this, _1, _2));
-            } else {
-                LogPrintf("tor: Password authentication required, but no password provided with -torpassword\n");
-            }
+            LogPrintf("tor: Only hashed password authentication supported, but no password provided with -torpassword\n");
```
Comment (sipa) [inline src/torcontrol.cpp]: This seems a bit misleading: couldn't there be supported authentication mechanism left which we don't support? Suggestion: "Only supported remaining authenthentication mechanism left is password, but ..."

**Comment by sipa** [comment]
Concept ACK

**Comment by gmaxwell** [comment]
Concept ACK

File: src/torcontrol.cpp (hunk 1 above)
Comment (laanwj) [inline src/torcontrol.cpp]: Agreed, will change the message

**Comment by sipa** [comment]
utACK 2e494489c35499dbe7891d4c07b4c35b178341d1

**Pull request merged**